tests/json-fixtures/test_transactions.py

```diff
@@ -40,43 +45,92 @@
 BASE_FIXTURE_PATH = os.path.join(ROOT_PROJECT_DIR, 'fixtures', 'TransactionTests')
 
 
+@to_tuple
+def expand_fixtures_forks(all_fixtures):
+    """
+    The transaction fixtures have different definitions for each fork and must be
+    expanded one step further to have one fixture for each defined fork within
+    the fixture.
+    """
+    # TODO: This was copied from state tests. Make more DRY.
+    for fixture_path, fixture_key in all_fixtures:
+        fixture = load_fixture(fixture_path, fixture_key)
+        for fixture_fork, fork_states in fixture.items():
+            if (fixture_fork in ForkName.__members__):
+                yield fixture_path, fixture_key, fixture_fork
+
+
 def pytest_generate_tests(metafunc):
     generate_fixture_tests(
         metafunc=metafunc,
         base_fixture_path=BASE_FIXTURE_PATH,
+        preprocess_fn=expand_fixtures_forks,
     )
 
 
 @pytest.fixture
 def fixture(fixture_data):
-    fixture_path, fixture_key = fixture_data
+    fixture_path, fixture_key, fixture_fork = fixture_data
     fixture = load_fixture(
         fixture_path,
         fixture_key,
-        normalize_transactiontest_fixture,
+        normalize_transactiontest_fixture(fork=fixture_fork),
     )
+
     return fixture
 
 
-def test_transaction_fixtures(fixture):
-    header = BlockHeader(1, fixture['blocknumber'], 100)
-    chain = MainnetChain(ChainDB(get_db_backend()), header=header)
-    vm = chain.get_vm()
-    TransactionClass = vm.get_transaction_class()
+@pytest.fixture
+def fixture_transaction_class(fixture_data):
+    _, _, fork_name = fixture_data
+    if fork_name == ForkName.Frontier.name:
+        return FrontierTransaction
+    elif fork_name == ForkName.Homestead.name:
+        return HomesteadTransaction
+    elif fork_name == ForkName.EIP150.name:
+        return HomesteadTransaction
+    elif fork_name == ForkName.EIP158.name:
+        return SpuriousDragonTransaction
+    elif fork_name == ForkName.Byzantium.name:
+        return ByzantiumTransaction
+    elif fork_name == ForkName.Sharding.name:
+        pytest.skip("Sharding Transaction class has not been implemented")
+    elif fork_name == ForkName.Constantinople.name:
+        pytest.skip("Constantinople Transaction class has not been implemented")
+    elif fork_name == ForkName.Metropolis.name:
+        pytest.skip("Metropolis Transaction class has not been implemented")
+    else:
+        raise ValueError("Unknown Fork Name: {0}".format(fork_name))
+
+
+def test_transaction_fixtures(fixture, fixture_transaction_class):
+
+    # TODO: Clarify the following:
```

Comment:
@pipermerriam previous comment got lost by my latest force push so I'm readding. Mind chiming in here to comment on my comment about the missing block number?